cypress/integration/value.cy.js

```diff
@@ -0,0 +1,71 @@
+import { DIMENSION_ID_EVENT_DATE } from '../../src/modules/dimensionConstants.js'
+import {
+    E2E_PROGRAM,
+    TEST_DIM_TEXT,
+    TEST_DIM_TEXT_OPTIONSET,
+    TEST_DIM_NUMBER_OPTIONSET,
+    TEST_DIM_NUMBER,
+    TEST_DIM_INTEGER,
+    TEST_DIM_INTEGER_POSITIVE,
+    TEST_DIM_INTEGER_NEGATIVE,
+    TEST_DIM_INTEGER_ZERO_OR_POSITIVE,
+    TEST_DIM_PERCENTAGE,
+    TEST_DIM_UNIT_INTERVAL,
+    TEST_DIM_TIME,
+    TEST_DIM_DATE,
+    TEST_DIM_DATETIME,
+    TEST_DIM_PHONE_NUMBER,
+    TEST_REL_PE_LAST_12_MONTHS,
+} from '../data/index.js'
+import { selectEventWithProgramDimensions } from '../helpers/dimensions.js'
+import { clickMenubarUpdateButton } from '../helpers/menubar.js'
+import { selectRelativePeriod } from '../helpers/period.js'
+import { getTableDataCells } from '../helpers/table.js'
+import { EXTENDED_TIMEOUT } from '../support/util.js'
+
+describe('value', () => {
+    beforeEach(() => {
+        cy.visit('/', EXTENDED_TIMEOUT)
+
+        selectEventWithProgramDimensions({
+            programName: E2E_PROGRAM.programName,
+            dimensions: [
+                TEST_DIM_TEXT,
+                TEST_DIM_TEXT_OPTIONSET,
+                TEST_DIM_NUMBER_OPTIONSET,
+                TEST_DIM_NUMBER,
+                TEST_DIM_INTEGER,
+                TEST_DIM_INTEGER_POSITIVE,
+                TEST_DIM_INTEGER_NEGATIVE,
+                TEST_DIM_INTEGER_ZERO_OR_POSITIVE,
+                TEST_DIM_PERCENTAGE,
+                TEST_DIM_UNIT_INTERVAL,
+                TEST_DIM_TIME,
+                TEST_DIM_DATE,
+                TEST_DIM_DATETIME,
+                TEST_DIM_PHONE_NUMBER,
+            ],
+        })
+
+        selectRelativePeriod({
+            label: E2E_PROGRAM[DIMENSION_ID_EVENT_DATE],
+            period: TEST_REL_PE_LAST_12_MONTHS,
+        })
+
+        clickMenubarUpdateButton()
+    })
+
+    it('has the correct white-space css', () => {
+        const shouldHaveWhiteSpace = (index, value) =>
+            getTableDataCells()
+                .eq(index)
+                .invoke('css', 'white-space')
+                .should('equal', value)
+
+        console.log(getTableDataCells().all())
+        ;[0, 1, 2, 3].forEach((index) => shouldHaveWhiteSpace(index, 'normal'))
```

Comment:
It's really difficult to understand why `0, 1, 2, 3` should be treated differently from the rest, especially since the actual table is generated in the `beforeEach` so it's quite disconnected from what's going on here.
Also tying the exact index of items in an array to the test result makes it quite fragile (adding something to the list above will mess up the test).

I would suggest to remove the `beforeEach` and keep everything within the `it`. Then, instead of relying on the exact index value, create a list of all dimensions that should have `nowrap` and all with `wrap`, then use `indexOf` to find them in the list.. 

e.g.

```

const dimensionsWithWrap = [TEST_DIM_TEXT, TEST_DIM_TEXT_OPTIONSET, TEST_DIM_NUMBER_OPTIONSET]
const dimensionsWithoutWrap = [......]
const dimensions =  [...dimensionsWithWrap, ...dimensionsWithoutWrap]

selectEventWithProgramDimensions({
            programName: E2E_PROGRAM.programName,
            dimensions,
...


dimensionsWithWrap.forEach((dimension) => shouldHaveWhiteSpace(dimensions.indexOf(dimension), 'normal'))
dimensionsWithoutWrap.forEach((dimension) => shouldHaveWhiteSpace(dimensions.indexOf(dimension), 'nowrap'))

```

This way you're not relying on exact indexes (less fragile) and the testes are much easier to read and understand what's going on.

Reply:
I wanted to include the orgunit and event date, which are not passed as dimensions to the select.. function. But the orgunit is just TEXT and event date is DATE which we already test, so we can skip those which makes it less complex.